processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -159,8 +159,18 @@ public GroupByQueryQueryToolChest(
 
       final Sequence<Row> subqueryResult = mergeGroupByResults(subquery, runner, context);
       final List<AggregatorFactory> aggs = Lists.newArrayList();
+
       for (AggregatorFactory aggregatorFactory : query.getAggregatorSpecs()) {
-        aggs.addAll(aggregatorFactory.getRequiredColumns());
```

Comment:
Does this get confused if someone wanted an outer query with two aggregators that hit the same underlying field in different ways? Like, two JavaScript aggregators that both reference a "value" field, but one of them returns 2*value and one returns 3*value.